gnucash/import-export/aqb/gnc-ab-gettrans.c

```diff
@@ -143,6 +143,23 @@ gnc_ab_gettrans(GtkWidget *parent, Account *gnc_acc)
         g_debug("gnc_ab_gettrans: gettrans_dates aborted");
         goto cleanup;
     }
+
+    /* validate date range: from_date should be less than or equal to to_date. */
+    if(GWEN_Time_toTime_t(from_date) > GWEN_Time_toTime_t(to_date))
+    {
+        g_debug("gnc_ab_gettrans: invalid date range - from_date is greater than to_date");
+        gnc_error_dialog (GTK_WINDOW (parent), _("Invalid date range: \nStart date cannot be greater than the End date!"));
+        goto cleanup;
+    }
+
+    /* validate to_date: to_date cann't be in the future */
+    if(GWEN_Time_toTime_t(to_date) > gnc_time(NULL))
+    {
+        g_debug("gnc_ab_gettrans: invalid date range - to_date is in the future");
+        gnc_error_dialog (GTK_WINDOW (parent), _("Invalid date range: \nEnd date cannot be in the future!"));
```

Comment:
Is a date in the future effectively causing a bank error ?

Reply:
Yes, in cases with the invalid date range GnuCash tries to get the connection to the bank and the bank send an error messages. It looks in GnuCash like this:
![bildschirmfoto_2018-07-24_20-23-54](https://user-images.githubusercontent.com/426910/43485695-3570409c-9512-11e8-96a0-92f856c124bc.png)

If I repeate the same requests on the console with `aqbanking-cli` I get the following error codes:

**"to-date" > today** or **"to-date" > "from-date"**:
```
...
Dialog not aborted, assuming PIN is ok
AqHBCI finished.
5:2018/07/31 22-45-28:aqbanking(3915):banking_online.c:  119: Error executing backend's queue
4:2018/07/31 22-45-28:aqbanking(3915):banking_online.c:  137: Not a single job successfully executed
Postprocessing jobs
Job Get Transactions: error
Resetting provider queues
Executing Jobs: Finished.

```